Ilya/NetflixRoulette/netflix-roulette/src/pages/home/HomePage.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import queryString from "query-string";
+import { RouteComponentProps } from "react-router-dom";
+
+import "./HomePage.scss"
+import Header from 'shared/header/Header';
+import SearchMovie from "shared/searchMovie/SearchMovie";
+import Footer from 'shared/footer/Footer';
+import MovieList from 'shared/movieList/MovieList';
+import Pagination from 'shared/pagination/Pagination';
+import MovieSorter, { SortBy } from "shared/movieSorter/MovieSorter";
+import { HomePageConnectProps } from '.';
+
+type HomePageProps = RouteComponentProps & HomePageConnectProps;
+
+class HomePage extends React.Component<HomePageProps> {
+
+    componentDidMount() {
+        this.props.requestMovies(this.props.location.search);
+    }
+
+    componentWillUnmount() {
+        this.props.resetMovies();
+    }
+
+    componentDidUpdate(prevProps: Readonly<HomePageProps>) {
+
+        if (this.props.history.action !== "PUSH" && this.props.location !== prevProps.location) {
+            this.props.requestMovies(this.props.location.search);
+        }
+    }
+
+    sortMovieByHandler = (sortBy: SortBy) => {
+        const parsed = queryString.parse(this.props.location.search);
+        const newSearchParams = queryString.stringify({ ...parsed, sortBy });
+        this.props.history.push(`/search?${newSearchParams}`);
+
+        this.props.requestMovies(newSearchParams);
+    }
+
+    submitSearchHandler = (search: string, searchBy: string) => {
+        const parsed = queryString.parse(this.props.location.search);
+        const newSearchParams = queryString.stringify({ ...parsed, search, searchBy });
+        this.props.history.push(`/search?${newSearchParams}`);
+
+        this.props.requestMovies(newSearchParams);
+    }
+
+    //Pagination Methods
+    changePage = (e: React.MouseEvent<HTMLButtonElement>) => {
+        let page;
+        const elem: HTMLButtonElement = e.target as HTMLButtonElement;
+
+        if (elem.value === "next" && this.props.currentPage < Math.ceil(this.props.total / this.props.limit)) {
+            page = this.props.currentPage + 1;
+            this.props.setCurrentPage(page);
```

Comment:
You understood me wrong. You have to leave logic of getting the page number inside of pagination and call some this.props.onPageChange(pageNumber) prop that you pass from Home component; Your Home component shouldn't do the calculation of page.  

SAME BELOW